keanu-project/src/test/java/io/improbable/keanu/vertices/dbl/nonprobabilistic/DoubleIfVertexTest.java

```diff
@@ -170,6 +170,40 @@ public void canExtractDualNumberFromMixedPredicateWithDifferentParentsAndFillInZ
         Assert.assertArrayEquals(dFdd.getShape(), dIfdD.getShape());
     }
 
+    @Test
+    public void canExtractDualNumberFromMixedPredicateWithDifferentParentsAndFillInZeroesRankThree() {
```

Comment:
What a name!

Could we perhaps cut it down a little bit though?  It's a bit of a mouthful for a test that is just checking that the shapes are correct?

Reply:
Ok - made it a little shorter!!